src/widget/modal.svelte

```diff
@@ -0,0 +1,277 @@
+<script context="module" lang="ts">
```

Comment:
I can see your earlier comment on inline.svelte now - I understand that this will be revisited in the future to reduce duplication, so this is not an issue

Reply:
Yeah, this is related to this comment: https://github.com/cerebrl/forgerock-web-login-framework/pull/9/files#r934712797.

I recognize the duplication of code, but pulling it out into a reusable utility will take a while. There's a lot of closures that are referenced here and preserving all of this in a reusable way will take a bit. What's more, I didn't want to delay the testing of a lot of these assumption as well.